containers/fedora/build.sh

```
@@ -71,22 +71,59 @@ if [ $FEDORA_VERSION -gt 40 ]; then
    OS_VARIANT="fedora40"
 fi
 
+# Work on a copy of the base image so the original is never mutated.
+# This ensures cloud-init always runs fresh on repeated executions.
+WORK_IMAGE="${FEDORA_IMAGE%.qcow2}-work.qcow2"
+cp "${FEDORA_IMAGE}" "${WORK_IMAGE}"
+
+
+# Clean up any leftovers from a previous failed run
+rm -rf $BUILD_DIR
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)